jit-service/src/main/java/org/dcsa/jit/service/TransportCallService.java

```diff
@@ -34,11 +35,18 @@ public class TransportCallService {
   @Transactional
   public TransportCall ensureTransportCallExists(TimestampTO timestampTO) {
 
-    if ((timestampTO.importVoyageNumber() != null && timestampTO.exportVoyageNumber() != null)
-      == (timestampTO.carrierVoyageNumber() != null)) {
-      throw ConcreteRequestErrorMessageException.invalidInput(
-        "Cannot create timestamp where both voyage numbers carrierVoyageNumber AND (exportVoyageNumber + importVoyageNumber) " +
-          "is missing or provided together");
+    if (timestampTO.importVoyageNumber() != null || timestampTO.exportVoyageNumber() != null) {
+      if (timestampTO.importVoyageNumber() == null || timestampTO.exportVoyageNumber() == null) {
+        throw ConcreteRequestErrorMessageException.invalidInput(
+            "Both voyage exportVoyageNumber + importVoyageNumber should either be provided together or not at all"
+                + "is missing or provided together");
+      }
+      if (!(timestampTO.carrierVoyageNumber().equals(timestampTO.importVoyageNumber())
+          || timestampTO.carrierVoyageNumber().equals(timestampTO.exportVoyageNumber()))) {
+        throw ConcreteRequestErrorMessageException.invalidInput(
+            "When importVoyageNumber & exportVoyageNumber is given, then one of them has to equal the carrierVoyageNumber."
+                + "Please verify the values");
+      }
```

Comment:
You are missing if all 3 are null - that should produce an error

Reply:
ahh - disregard this comment - you have added the @notnull annotation on the model class